### PR TITLE
feat: [DHIS2-14778][DHIS2-14779][DHIS2-14780][DHIS2-14781] MULTI_TEXT field

### PR DESCRIPTION
**Pull request by simonadomnisoru**
[DHIS2-14778](https://dhis2.atlassian.net/browse/DHIS2-14778)

**Tech summary**
The multi-text values are converted and sent to the server. The saved values are converted and displayed in the edit mode of all our forms [DHIS2-14778](https://dhis2.atlassian.net/browse/DHIS2-14778):
- single events (new and edit)  
- enrollment events (new and edit)  
- TEI (new and edit)  
- new TEI when adding a relationship
- search form (due to API constraints to search only one value at that time, I reuse the existing dropdown UI )
- custom forms

The value is converted and displayed in the:
- view mode of the forms and the profile widget [DHIS2-14780](https://dhis2.atlassian.net/browse/DHIS2-14780) 
- working lists [DHIS2-14779](https://dhis2.atlassian.net/browse/DHIS2-14779) 
- stage and events widget [DHIS2-14781](https://dhis2.atlassian.net/browse/DHIS2-14781) 

I followed as much as possible the existing OptionSet form field logic. 

**Next steps**
The rules engine logic will be done in [DHIS2-13436](https://dhis2.atlassian.net/browse/DHIS2-13436), which is currently blocked by the backend [DHIS2-13435](https://dhis2.atlassian.net/browse/DHIS2-13435) 

[DHIS2-13436]: https://dhis2.atlassian.net/browse/DHIS2-13436?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-13435]: https://dhis2.atlassian.net/browse/DHIS2-13435?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DHIS2-14778]: https://dhis2.atlassian.net/browse/DHIS2-14778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-14780]: https://dhis2.atlassian.net/browse/DHIS2-14780?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-14779]: https://dhis2.atlassian.net/browse/DHIS2-14779?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DHIS2-14781]: https://dhis2.atlassian.net/browse/DHIS2-14781?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JoakimSM — COMMENTED** [review]
Thanks @simonadomnisoru, Looks very good overall!

- At the moment you can create data elements / tracked entity attributes of type MULIT_TEXT with and without associated option sets. In our app we are only planning to support MULTI_TEXT with associated option sets. Let’s do some validation here. I suggest we validate when we create the metadata memory objects. If a MULTI_TEXT without an associated option set is found, we disregard the the data element / tracked entity attribute and log an error to the console. 

- Great if we can remove «Virtualized» from any multi select field components. This should never have been used for single select either (it was just referring to implementation details in the underlying components)

**Comment by eirikhaugstulen** [comment]
Hey @simonadomnisoru,
This looks really good! One small comment:
It looks like we use the codes instead of the displayed list in the top bar. Is this by design?

image

**Comment by JoakimSM** [comment]
@simonadomnisoru: After you have looked into @eirikhaugstulen's comment and my nitpicking one, we can approve this PR.

**Comment by simonadomnisoru** [comment]
> Hey @simonadomnisoru, This looks really good! One small comment: It looks like we use the codes instead of the displayed list in the top bar. Is this by design?
> 
> image

Hey @eirikhaugstulen,
Good catch! I committed a fix for this. Let me know if you find any more issues. 
Thanks for the feedback 😄

File: src/core_modules/capture-core/components/D2Form/field/Components/MultiOptionSetField/withRulesOptionVisibilityHandler.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+// @flow
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/D2Form/field/Components/MultiOptionSetField/withRulesOptionVisibilityHandler.js]: @JoakimSM the file was moved to the folder you suggested. Thanks for the review!

**Review by JoakimSM — APPROVED** [review]

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.41 version

**Pull request merged**